Components/DeckCard.js

```diff
@@ -17,13 +17,8 @@ const DeckCard = props => {
 
 function mapStateToProps(decks, { title }) {
   //for when the there is no cards
-  let numberOfCards = 0;
-  if (decks[title].questions !== undefined && title !== undefined) {
-    numberOfCards = decks[title].questions.length;
-  }
-
   return {
-    numberOfCards
+    numberOfCards: decks[title] ? decks[title].questions : 0
```

Comment:
Here is the main fix of the error. You were trying to access the property `questions` of a possibly undefined object `deck[titles]`. With this new implementation, we check for the existence of the object first. If it exists, we're taking the length of the array `questions`, and if undefined, we return the value `0`.